### PR TITLE
Concourse pipeline for codegen

### PR DESCRIPTION
**Pull request by hardikar**
* Choosing Centos 6 because that's our de-facto standard at this point in time
* Refactored some of the python scripts.
* Added `build/test_with_orca/codegen/orca_and_codegen.yml` files
* Added the pipeline in the GPDB code base.
* Added `prepare_orca_dependencies` for wget-ing the right ORCA/GPOS binaries based on ivy.xml file. (Until a better solution is available)

### CODE REVIEW

**Comment by hardikar** [comment]
Currently running pipeline (on my branch) [here](https://gporca.ci.pivotalci.info/pipelines/gpcodegen_test). I will update that to use gpcodegen instead of gpcodegen_test when this PR is pushed in.

**Comment by d** [comment]
Looks good on first glance. Can I `fly execute` things against your pipeline later today?

**Comment by hardikar** [comment]
Go for it.

File: concourse/test_with_orca_and_codegen.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+inputs:
+  - name: gpdb_src
+  - name: bin_gpdb
+  - name: bin_orca
+  - name: bin_gpos
+  - name: bin_xerces
+run:
+  path: /bin/bash
+  args:
+    - -l
```
Comment (gcaragea) [inline concourse/test_with_orca_and_codegen.yml]: Please add a comment here on why using -l is crucial.

Comment (d) [inline concourse/test_with_orca_and_codegen.yml]: You want a login shell when you care about some login shell settings, which is a smell, but the intent is clear I believe.

File: concourse/test_with_orca_and_codegen.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    - -c
+    - gpdb_src/concourse/scripts/test_gpdb.py --mode=orca_codegen --gpdb_name=bin_gpdb --output_dir=bin_gpdb/install bin_orca bin_gpos bin_xerces
+
```
Comment (d) [inline concourse/test_with_orca_and_codegen.yml]: I'm surprised you still pull in `bin_orca` given that you don't use it as an input

Comment (hardikar) [inline concourse/test_with_orca_and_codegen.yml]: I do - from prepeare_orca_dependencies - This makes sure that the
directories are created. But they're not resources.

**Comment by d** [comment]
+1

**Pull request merged**